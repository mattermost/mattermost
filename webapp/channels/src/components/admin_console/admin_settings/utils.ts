// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type {AdminConfig, EnvironmentConfig} from '@mattermost/types/config';

function getConfigValue(config: AdminConfig | Partial<EnvironmentConfig>, path: string) {
    const pathParts = path.split('.');

    return pathParts.reduce((obj: object | null, pathPart) => {
        if (!obj) {
            return null;
        }
        return obj[(pathPart as keyof object)];
    }, config);
}

export function isSetByEnv(environmentConfig: Partial<EnvironmentConfig>, path: string) {
    return Boolean(getConfigValue(environmentConfig, path));
}

export const parseIntNonNegative = (str: string | number, defaultValue?: number) => {
    const n = typeof str === 'string' ? parseInt(str, 10) : str;

    if (isNaN(n) || n < 0) {
        if (defaultValue) {
            return defaultValue;
        }
        return 0;
    }

    return n;
};

export const parseIntZeroOrMin = (str: string | number, minimumValue = 1) => {
    const n = typeof str === 'string' ? parseInt(str, 10) : str;

    if (isNaN(n) || n < 0) {
        return 0;
    }
    if (n > 0 && n < minimumValue) {
        return minimumValue;
    }

    return n;
};

export const parseIntNonZero = (str: string | number, defaultValue?: number, minimumValue = 1) => {
    const n = typeof str === 'string' ? parseInt(str, 10) : str;

    if (isNaN(n) || n < minimumValue) {
        if (defaultValue) {
            return defaultValue;
        }
        return 1;
    }

    return n;
};
