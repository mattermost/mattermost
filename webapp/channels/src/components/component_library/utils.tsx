// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

function buildPropsLists(inputPossibilities: {[x: string]: any[]}): Array<{[x: string]: any}> {
    const keys = Object.keys(inputPossibilities);
    if (!keys.length) {
        return [{}];
    }

    const selectedKey = keys[0];
    const restPossibilities = {...inputPossibilities};
    delete restPossibilities[selectedKey];
    const subProps = buildPropsLists(restPossibilities);
    const result: Array<{[x: string]: any}> = [];
    inputPossibilities[selectedKey].forEach((v) => {
        subProps.forEach((rest) => {
            result.push({...rest, [selectedKey]: v});
        });
    });

    return result;
}

function buildPropString(inputProps: {[x: string]: any}) {
    const propKeys = Object.keys(inputProps);
    if (!propKeys.length) {
        return undefined;
    }

    const result = [(<>{'PROPS: '}</>)];
    propKeys.forEach((v) => {
        result.push((<><b>{v}</b>{`: ${inputProps[v]}, `}</>));
    });
    return result;
}

export function buildComponent(
    Component: React.ComponentType<any>,
    propPossibilities: {[x: string]: any[]},
    dropdownPossibilities: Array<{[x: string]: string[]} | undefined>,
    setProps: Array<{[x: string]: any} | undefined>,
) {
    const res: React.ReactNode[] = [];
    let currentPropPossibilities = {...propPossibilities};
    dropdownPossibilities.forEach((v) => {
        if (v) {
            currentPropPossibilities = {
                ...currentPropPossibilities,
                ...v,
            };
        }
    });

    const propsVariations = buildPropsLists(currentPropPossibilities);
    let builtSetProps = {};
    setProps.forEach((v) => {
        if (v) {
            builtSetProps = {
                ...builtSetProps,
                ...v,
            };
        }
    });
    propsVariations.forEach((v) => {
        const propString = buildPropString(v);
        res.push(
            <>
                {Boolean(propString) && <p>{propString}</p>}
                <Component
                    {...builtSetProps}
                    {...v}
                />
            </>,
        );
    });
    return res;
}
