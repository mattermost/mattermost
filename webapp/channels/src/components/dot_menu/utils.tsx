// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type React from 'react';

export type ChangeEvent = React.KeyboardEvent | React.MouseEvent;

export const trackDotMenuEvent = (): void => {
    // Telemetry tracking removed
};
