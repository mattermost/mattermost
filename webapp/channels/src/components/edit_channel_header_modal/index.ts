// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type {ConnectedProps} from 'react-redux';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import type {Dispatch} from 'redux';

import {patchChannel} from 'mattermost-redux/actions/channels';
import {Preferences} from 'mattermost-redux/constants';
import {getBool} from 'mattermost-redux/selectors/entities/preferences';

import {setShowPreviewOnEditChannelHeaderModal} from 'actions/views/textbox';
import {showPreviewOnEditChannelHeaderModal} from 'selectors/views/textbox';

import type {GlobalState} from 'types/store';

import EditChannelHeaderModal from './edit_channel_header_modal';

function mapStateToProps(state: GlobalState) {
    return {
        shouldShowPreview: showPreviewOnEditChannelHeaderModal(state),
        ctrlSend: getBool(state, Preferences.CATEGORY_ADVANCED_SETTINGS, 'send_on_ctrl_enter'),
    };
}

function mapDispatchToProps(dispatch: Dispatch) {
    return {
        actions: bindActionCreators({
            patchChannel,
            setShowPreview: setShowPreviewOnEditChannelHeaderModal,
        }, dispatch),
    };
}

const connector = connect(mapStateToProps, mapDispatchToProps);

export type PropsFromRedux = ConnectedProps<typeof connector>;

export default connect(mapStateToProps, mapDispatchToProps)(EditChannelHeaderModal);
