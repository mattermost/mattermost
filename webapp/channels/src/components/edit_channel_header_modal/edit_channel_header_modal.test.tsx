// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';

import type {Channel, ChannelType} from '@mattermost/types/channels';

import {EditChannelHeaderModal} from 'components/edit_channel_header_modal/edit_channel_header_modal';
import type {EditChannelHeaderModal as EditChannelHeaderModalClass} from 'components/edit_channel_header_modal/edit_channel_header_modal';
import Textbox from 'components/textbox';

import {type MockIntl} from 'tests/helpers/intl-test-helper';
import {testComponentForLineBreak} from 'tests/helpers/line_break_helpers';
import Constants from 'utils/constants';
import * as Utils from 'utils/utils';

const KeyCodes = Constants.KeyCodes;

describe('components/EditChannelHeaderModal', () => {
    const timestamp = Utils.getTimestamp();
    const channel = {
        id: 'fake-id',
        create_at: timestamp,
        update_at: timestamp,
        delete_at: timestamp,
        team_id: 'fake-team-id',
        type: Constants.OPEN_CHANNEL as ChannelType,
        display_name: 'Fake Channel',
        name: 'Fake Channel',
        header: 'Fake Channel',
        purpose: 'purpose',
        last_post_at: timestamp,
        creator_id: 'fake-creator-id',
        scheme_id: 'fake-scheme-id',
        group_constrained: false,
        last_root_post_at: timestamp,
    };

    const serverError = {
        server_error_id: 'fake-server-error',
        message: 'some error',
    };

    const baseProps = {
        markdownPreviewFeatureIsEnabled: false,
        channel,
        ctrlSend: false,
        show: false,
        shouldShowPreview: false,
        onExited: jest.fn(),
        actions: {
            setShowPreview: jest.fn(),
            patchChannel: jest.fn().mockResolvedValue({}),
        },
        intl: {
            formatMessage: ({defaultMessage}) => defaultMessage,
        } as MockIntl,
    };

    test('should match snapshot, init', () => {
        const wrapper = shallow(
            <EditChannelHeaderModal {...baseProps}/>,
        );
        expect(wrapper).toMatchSnapshot();
    });
    test('edit direct message channel', () => {
        const dmChannel: Channel = {
            ...channel,
            type: Constants.DM_CHANNEL as ChannelType,
        };

        const wrapper = shallow(
            <EditChannelHeaderModal
                {...baseProps}
                channel={dmChannel}
            />,
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('submitted', () => {
        const wrapper = shallow(
            <EditChannelHeaderModal {...baseProps}/>,
        );

        wrapper.setState({saving: true});

        expect(wrapper).toMatchSnapshot();
    });

    test('error with intl message', () => {
        const wrapper = shallow(
            <EditChannelHeaderModal {...baseProps}/>,
        );

        wrapper.setState({serverError: {...serverError, server_error_id: 'model.channel.is_valid.header.app_error'}});
        expect(wrapper).toMatchSnapshot();
    });

    test('error without intl message', () => {
        const wrapper = shallow(
            <EditChannelHeaderModal {...baseProps}/>,
        );

        wrapper.setState({serverError});
        expect(wrapper).toMatchSnapshot();
    });

    describe('handleSave', () => {
        test('on no change, should hide the modal without trying to patch a channel', async () => {
            const wrapper = shallow(
                <EditChannelHeaderModal {...baseProps}/>,
            );

            const instance = wrapper.instance() as EditChannelHeaderModalClass;

            await instance.handleSave();

            expect(wrapper.state('show')).toBe(false);

            expect(baseProps.actions.patchChannel).not.toHaveBeenCalled();
        });

        test('on error, should not close modal and set server error state', async () => {
            baseProps.actions.patchChannel.mockResolvedValueOnce({error: serverError});

            const wrapper = shallow(
                <EditChannelHeaderModal {...baseProps}/>,
            );

            const instance = wrapper.instance() as EditChannelHeaderModalClass;

            wrapper.setState({header: 'New header'});

            await instance.handleSave();

            expect(wrapper.state('show')).toBe(true);
            expect(wrapper.state('serverError')).toBe(serverError);

            expect(baseProps.actions.patchChannel).toHaveBeenCalled();
        });

        test('on success, should close modal', async () => {
            const wrapper = shallow(
                <EditChannelHeaderModal {...baseProps}/>,
            );

            const instance = wrapper.instance() as EditChannelHeaderModalClass;

            wrapper.setState({header: 'New header'});

            await instance.handleSave();

            expect(wrapper.state('show')).toBe(false);

            expect(baseProps.actions.patchChannel).toHaveBeenCalled();
        });
    });

    test('change header', () => {
        const wrapper = shallow(
            <EditChannelHeaderModal {...baseProps}/>,
        );

        wrapper.find(Textbox).simulate('change', {target: {value: 'header'}});

        expect(
            wrapper.state('header'),
        ).toBe('header');
    });

    test('patch on save button click', () => {
        const wrapper = shallow(
            <EditChannelHeaderModal {...baseProps}/>,
        );

        const newHeader = 'New channel header';
        wrapper.setState({header: newHeader});
        wrapper.find('.save-button').simulate('click');

        expect(baseProps.actions.patchChannel).toHaveBeenCalledWith('fake-id', {header: newHeader});
    });

    test('patch on enter keypress event with ctrl', () => {
        const wrapper = shallow(
            <EditChannelHeaderModal
                {...baseProps}
                ctrlSend={true}
            />,
        );

        const newHeader = 'New channel header';
        wrapper.setState({header: newHeader});
        wrapper.find(Textbox).simulate('keypress', {
            preventDefault: jest.fn(),
            key: KeyCodes.ENTER[0],
            which: KeyCodes.ENTER[1],
            shiftKey: false,
            altKey: false,
            ctrlKey: true,
        });

        expect(baseProps.actions.patchChannel).toHaveBeenCalledWith('fake-id', {header: newHeader});
    });

    test('patch on enter keypress', () => {
        const wrapper = shallow(
            <EditChannelHeaderModal {...baseProps}/>,
        );

        const newHeader = 'New channel header';
        wrapper.setState({header: newHeader});
        wrapper.find(Textbox).simulate('keypress', {
            preventDefault: jest.fn(),
            key: KeyCodes.ENTER[0],
            which: KeyCodes.ENTER[1],
            shiftKey: false,
            altKey: false,
            ctrlKey: false,
        });

        expect(baseProps.actions.patchChannel).toHaveBeenCalledWith('fake-id', {header: newHeader});
    });

    test('patch on enter keydown', () => {
        const wrapper = shallow(
            <EditChannelHeaderModal
                {...baseProps}
                ctrlSend={true}
            />,
        );

        const newHeader = 'New channel header';
        wrapper.setState({header: newHeader});
        wrapper.find(Textbox).simulate('keydown', {
            preventDefault: jest.fn(),
            key: KeyCodes.ENTER[0],
            keyCode: KeyCodes.ENTER[1],
            which: KeyCodes.ENTER[1],
            shiftKey: false,
            altKey: false,
            ctrlKey: true,
        });

        expect(baseProps.actions.patchChannel).toHaveBeenCalledWith('fake-id', {header: newHeader});
    });

    test('should show error only for invalid length', () => {
        const wrapper = shallow(
            <EditChannelHeaderModal {...baseProps}/>,
        );

        wrapper.find(Textbox).simulate('change', {target: {value: `The standard Lorem Ipsum passage, used since the 1500s
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."

        Section 1.10.32 of "de Finibus Bonorum et Malorum", written by Cicero in 45 BC
        "Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt. Neque porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt ut labore et dolore magnam aliquam quaerat voluptatem. Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid ex ea commodi consequatur? Quis autem vel eum iure reprehenderit qui in ea voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem eum fugiat quo voluptas nulla pariatur?"

        1914 translation by H. Rackham
        "But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness. No one rejects, dislikes, or avoids pleasure itself, because it is pleasure, but because those who do not know how to pursue pleasure rationally encounter consequences that are extremely painful. Nor again is there anyone who loves or pursues or desires to obtain pain of itself, because it is pain, but because occasionally circumstances occur in which toil and pain can procure him some great pleasure. To take a trivial example, which of us ever undertakes laborious physical exercise, except to obtain some advantage from it? But who has any right to find fault with a man who chooses to enjoy a pleasure that has no annoying consequences, or one who avoids a pain that produces no resultant pleasure?"

        Section 1.10.33 of "de Finibus Bonorum et Malorum", written by Cicero in 45 BC
        "At vero eos et accusamus et iusto odio dignissimos ducimus qui blanditiis praesentium voluptatum deleniti atque corrupti quos dolores et quas molestias excepturi sint occaecati cupiditate non provident, similique sunt in culpa qui officia deserunt mollitia animi, id est laborum et dolorum fuga. Et harum quidem rerum facilis est et expedita distinctio. Nam libero tempore, cum soluta nobis est eligendi optio cumque nihil impedit quo minus id quod maxime placeat facere possimus, omnis voluptas assumenda est, omnis dolor repellendus. Temporibus autem quibusdam et aut officiis debitis aut rerum necessitatibus saepe eveniet ut et voluptates repudiandae sint et molestiae non recusandae. Itaque earum rerum hic tenetur a sapiente delectus, ut aut reiciendis voluptatibus maiores alias consequatur aut perferendis doloribus asperiores repellat."`}});

        expect(
            wrapper.state('serverError'),
        ).toStrictEqual({message: 'Invalid header length', server_error_id: 'model.channel.is_valid.header.app_error'});

        wrapper.find(Textbox).simulate('change', {target: {value: 'valid header'}});

        expect(
            wrapper.state('serverError'),
        ).toBeNull();
    });

    testComponentForLineBreak(
        (value: string) => (
            <EditChannelHeaderModal
                {...baseProps}
                channel={{
                    ...baseProps.channel,
                    header: value,
                }}
            />
        ),
        (instance: React.Component<any, any>) => instance.state.header,
        false,
    );
});
