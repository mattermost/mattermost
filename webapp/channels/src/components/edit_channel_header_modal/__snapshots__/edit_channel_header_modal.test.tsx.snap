// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/EditChannelHeaderModal edit direct message channel 1`] = `
<Modal
  animation={true}
  aria-labelledby="editChannelHeaderModalLabel"
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogClassName="a11y__modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={false}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onEntering={[Function]}
  onExited={[MockFunction]}
  onHide={[Function]}
  onKeyDown={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  role="none"
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h1"
      id="editChannelHeaderModalLabel"
    >
      <MemoizedFormattedMessage
        defaultMessage="Edit Header"
        id="edit_channel_header_modal.title_dm"
      />
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body edit-modal-body"
    componentClass="div"
  >
    <div>
      <p>
        <MemoizedFormattedMessage
          defaultMessage="Edit the text appearing next to the channel name in the header."
          id="edit_channel_header_modal.description"
        />
      </p>
      <div
        className="textarea-wrapper"
      >
        <Connect(Textbox)
          channelId="fake-id"
          characterLimit={1024}
          createMessage="Enter the Channel Header"
          handlePostError={[Function]}
          id="edit_textbox"
          onChange={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          preview={false}
          suggestionListPosition="bottom"
          supportsCommands={false}
          useChannelMentions={false}
          value="Fake Channel"
        />
      </div>
      <div
        className="post-create-footer"
      >
        <TextboxLinks
          hasExceededCharacterLimit={false}
          hasText={true}
          previewMessageLink={
            <Memo(MemoizedFormattedMessage)
              defaultMessage="Edit"
              id="edit_channel_header_modal.previewHeader"
            />
          }
          showPreview={false}
          updatePreview={[Function]}
        />
      </div>
    </div>
  </ModalBody>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
  >
    <button
      className="btn btn-tertiary cancel-button"
      onClick={[Function]}
      type="button"
    >
      <MemoizedFormattedMessage
        defaultMessage="Cancel"
        id="edit_channel_header_modal.cancel"
      />
    </button>
    <button
      className="btn btn-primary save-button"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <MemoizedFormattedMessage
        defaultMessage="Save"
        id="edit_channel_header_modal.save"
      />
    </button>
  </ModalFooter>
</Modal>
`;

exports[`components/EditChannelHeaderModal error with intl message 1`] = `
<Modal
  animation={true}
  aria-labelledby="editChannelHeaderModalLabel"
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogClassName="a11y__modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={false}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onEntering={[Function]}
  onExited={[MockFunction]}
  onHide={[Function]}
  onKeyDown={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  role="none"
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h1"
      id="editChannelHeaderModalLabel"
    >
      <MemoizedFormattedMessage
        defaultMessage="Edit Header for {channel}"
        id="edit_channel_header_modal.title"
        values={
          Object {
            "channel": "Fake Channel",
          }
        }
      />
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body edit-modal-body"
    componentClass="div"
  >
    <div>
      <p>
        <MemoizedFormattedMessage
          defaultMessage="Edit the text appearing next to the channel name in the header."
          id="edit_channel_header_modal.description"
        />
      </p>
      <div
        className="textarea-wrapper"
      >
        <Connect(Textbox)
          channelId="fake-id"
          characterLimit={1024}
          createMessage="Enter the Channel Header"
          handlePostError={[Function]}
          id="edit_textbox"
          onChange={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          preview={false}
          suggestionListPosition="bottom"
          supportsCommands={false}
          useChannelMentions={false}
          value="Fake Channel"
        />
      </div>
      <div
        className="post-create-footer"
      >
        <TextboxLinks
          hasExceededCharacterLimit={false}
          hasText={true}
          previewMessageLink={
            <Memo(MemoizedFormattedMessage)
              defaultMessage="Edit"
              id="edit_channel_header_modal.previewHeader"
            />
          }
          showPreview={false}
          updatePreview={[Function]}
        />
      </div>
      <div
        className="form-group has-error"
      >
        <br />
        <label
          className="control-label"
        >
          <MemoizedFormattedMessage
            defaultMessage="The text entered exceeds the character limit. The channel header is limited to {maxLength} characters."
            id="edit_channel_header_modal.error"
            values={
              Object {
                "maxLength": 1024,
              }
            }
          />
        </label>
      </div>
    </div>
  </ModalBody>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
  >
    <button
      className="btn btn-tertiary cancel-button"
      onClick={[Function]}
      type="button"
    >
      <MemoizedFormattedMessage
        defaultMessage="Cancel"
        id="edit_channel_header_modal.cancel"
      />
    </button>
    <button
      className="btn btn-primary save-button"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <MemoizedFormattedMessage
        defaultMessage="Save"
        id="edit_channel_header_modal.save"
      />
    </button>
  </ModalFooter>
</Modal>
`;

exports[`components/EditChannelHeaderModal error without intl message 1`] = `
<Modal
  animation={true}
  aria-labelledby="editChannelHeaderModalLabel"
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogClassName="a11y__modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={false}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onEntering={[Function]}
  onExited={[MockFunction]}
  onHide={[Function]}
  onKeyDown={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  role="none"
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h1"
      id="editChannelHeaderModalLabel"
    >
      <MemoizedFormattedMessage
        defaultMessage="Edit Header for {channel}"
        id="edit_channel_header_modal.title"
        values={
          Object {
            "channel": "Fake Channel",
          }
        }
      />
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body edit-modal-body"
    componentClass="div"
  >
    <div>
      <p>
        <MemoizedFormattedMessage
          defaultMessage="Edit the text appearing next to the channel name in the header."
          id="edit_channel_header_modal.description"
        />
      </p>
      <div
        className="textarea-wrapper"
      >
        <Connect(Textbox)
          channelId="fake-id"
          characterLimit={1024}
          createMessage="Enter the Channel Header"
          handlePostError={[Function]}
          id="edit_textbox"
          onChange={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          preview={false}
          suggestionListPosition="bottom"
          supportsCommands={false}
          useChannelMentions={false}
          value="Fake Channel"
        />
      </div>
      <div
        className="post-create-footer"
      >
        <TextboxLinks
          hasExceededCharacterLimit={false}
          hasText={true}
          previewMessageLink={
            <Memo(MemoizedFormattedMessage)
              defaultMessage="Edit"
              id="edit_channel_header_modal.previewHeader"
            />
          }
          showPreview={false}
          updatePreview={[Function]}
        />
      </div>
      <div
        className="form-group has-error"
      >
        <br />
        <label
          className="control-label"
        >
          some error
        </label>
      </div>
    </div>
  </ModalBody>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
  >
    <button
      className="btn btn-tertiary cancel-button"
      onClick={[Function]}
      type="button"
    >
      <MemoizedFormattedMessage
        defaultMessage="Cancel"
        id="edit_channel_header_modal.cancel"
      />
    </button>
    <button
      className="btn btn-primary save-button"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <MemoizedFormattedMessage
        defaultMessage="Save"
        id="edit_channel_header_modal.save"
      />
    </button>
  </ModalFooter>
</Modal>
`;

exports[`components/EditChannelHeaderModal should match snapshot, init 1`] = `
<Modal
  animation={true}
  aria-labelledby="editChannelHeaderModalLabel"
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogClassName="a11y__modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={false}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onEntering={[Function]}
  onExited={[MockFunction]}
  onHide={[Function]}
  onKeyDown={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  role="none"
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h1"
      id="editChannelHeaderModalLabel"
    >
      <MemoizedFormattedMessage
        defaultMessage="Edit Header for {channel}"
        id="edit_channel_header_modal.title"
        values={
          Object {
            "channel": "Fake Channel",
          }
        }
      />
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body edit-modal-body"
    componentClass="div"
  >
    <div>
      <p>
        <MemoizedFormattedMessage
          defaultMessage="Edit the text appearing next to the channel name in the header."
          id="edit_channel_header_modal.description"
        />
      </p>
      <div
        className="textarea-wrapper"
      >
        <Connect(Textbox)
          channelId="fake-id"
          characterLimit={1024}
          createMessage="Enter the Channel Header"
          handlePostError={[Function]}
          id="edit_textbox"
          onChange={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          preview={false}
          suggestionListPosition="bottom"
          supportsCommands={false}
          useChannelMentions={false}
          value="Fake Channel"
        />
      </div>
      <div
        className="post-create-footer"
      >
        <TextboxLinks
          hasExceededCharacterLimit={false}
          hasText={true}
          previewMessageLink={
            <Memo(MemoizedFormattedMessage)
              defaultMessage="Edit"
              id="edit_channel_header_modal.previewHeader"
            />
          }
          showPreview={false}
          updatePreview={[Function]}
        />
      </div>
    </div>
  </ModalBody>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
  >
    <button
      className="btn btn-tertiary cancel-button"
      onClick={[Function]}
      type="button"
    >
      <MemoizedFormattedMessage
        defaultMessage="Cancel"
        id="edit_channel_header_modal.cancel"
      />
    </button>
    <button
      className="btn btn-primary save-button"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <MemoizedFormattedMessage
        defaultMessage="Save"
        id="edit_channel_header_modal.save"
      />
    </button>
  </ModalFooter>
</Modal>
`;

exports[`components/EditChannelHeaderModal submitted 1`] = `
<Modal
  animation={true}
  aria-labelledby="editChannelHeaderModalLabel"
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogClassName="a11y__modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={false}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onEntering={[Function]}
  onExited={[MockFunction]}
  onHide={[Function]}
  onKeyDown={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  role="none"
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h1"
      id="editChannelHeaderModalLabel"
    >
      <MemoizedFormattedMessage
        defaultMessage="Edit Header for {channel}"
        id="edit_channel_header_modal.title"
        values={
          Object {
            "channel": "Fake Channel",
          }
        }
      />
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body edit-modal-body"
    componentClass="div"
  >
    <div>
      <p>
        <MemoizedFormattedMessage
          defaultMessage="Edit the text appearing next to the channel name in the header."
          id="edit_channel_header_modal.description"
        />
      </p>
      <div
        className="textarea-wrapper"
      >
        <Connect(Textbox)
          channelId="fake-id"
          characterLimit={1024}
          createMessage="Enter the Channel Header"
          handlePostError={[Function]}
          id="edit_textbox"
          onChange={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          preview={false}
          suggestionListPosition="bottom"
          supportsCommands={false}
          useChannelMentions={false}
          value="Fake Channel"
        />
      </div>
      <div
        className="post-create-footer"
      >
        <TextboxLinks
          hasExceededCharacterLimit={false}
          hasText={true}
          previewMessageLink={
            <Memo(MemoizedFormattedMessage)
              defaultMessage="Edit"
              id="edit_channel_header_modal.previewHeader"
            />
          }
          showPreview={false}
          updatePreview={[Function]}
        />
      </div>
    </div>
  </ModalBody>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
  >
    <button
      className="btn btn-tertiary cancel-button"
      onClick={[Function]}
      type="button"
    >
      <MemoizedFormattedMessage
        defaultMessage="Cancel"
        id="edit_channel_header_modal.cancel"
      />
    </button>
    <button
      className="btn btn-primary save-button"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      <MemoizedFormattedMessage
        defaultMessage="Save"
        id="edit_channel_header_modal.save"
      />
    </button>
  </ModalFooter>
</Modal>
`;
